src/WebAPI/Controllers/PlexAccountController.cs

```diff
@@ -115,26 +135,10 @@ public async Task<IActionResult> Validate([FromBody] PlexAccountDTO account)
     [ProducesResponseType(StatusCodes.Status500InternalServerError, Type = typeof(ResultDTO))]
     public async Task<IActionResult> CheckUsername(string username)
     {
-        if (string.IsNullOrEmpty(username) || username.Length < 5)
-            return BadRequest(Result.Fail("Invalid username"));
-
         try
         {
-            var result = await _plexAccountService.CheckIfUsernameIsAvailableAsync(username);
-
-            if (result.IsFailed)
-                return BadRequest(result.ToResult());
-
-            if (result.Value)
-            {
-                var logEvent = _log.Debug("Username: {UserName} is available", username);
-                return Ok(Result.Ok(true).WithSuccess(logEvent.ToLogString()));
-            }
-            else
-            {
-                var logEvent = _log.Warning("Account with username: {UserName} already exists!", username);
-                return Ok(Result.Ok(false).WithError(logEvent.ToLogString()));
-            }
+            var result = await _mediator.Send(new CheckUsernameTaskRequest(username));
```

Comment:
Now that you reduced the code of this endpoint, pay attention to what changed.
You only use the `mediator` dependency now, but your controller got 5 dependencies injected. It might be a good idea to take a look at FastEndpoints to make sure a single endpoints only get what it needs (SRP)

Reply:
Yes, good one! I came accross FastEndPoints today as well and it looked great, so should I copy the file structure to the API project and create endpoint files for each "feature/use case" that needs it? So in this case: `[WebAPI]/PlexAccounts/CheckUsername/CheckUsernameEndpoint.cs` or is it better to place this endpoint in the `Application` project to keep it more grouped?

Reply:
Exactly 😄 Now, look at the potential content of this folder:
- Dedicated endpoint
- Query
- Handler
You managed to go to a point where (almost - I'm getting back to that later) everything about this use case is in that folder. This was our target: high cohesion. It's suddenly easier to understand the responsibilities of this use case, and you can isolate the work.

Two more discussions emerge:
- Your endpoint suddenly sends a single message (CheckUsernameTaskRequest) over mediator. It does question the necessity of mediator as you could replace it by the handler directly. I'm not saying you 100% should, but it's worth investigating.
- Inside your handler, you're still sending a message to another handler. It's worth bringing that logic too (and maybe not rely on mediator for it). If you do it, you get your verticle slice because this single folder contains all the flow from top to bottom (or end to end)